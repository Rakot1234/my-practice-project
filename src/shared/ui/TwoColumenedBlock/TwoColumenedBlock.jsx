import React, { Component } from 'react';
import './TwoColumenedBlock.scss';
import cx from 'classnames';
import PropTypes from 'prop-types';

class TwoColumenedBlock extends Component {
    static propTypes = {
        left: PropTypes.node,
        right: PropTypes.node
    };

    render() {
        const { left, right } = this.props;

        return (
            <div className={cx('two-columened-block')}>
                <div className={cx('two-columened-block__left')}>
                    {left}
                </div>
                <div className={cx('two-columened-block__right')}>
                    {right}
                </div>
            </div>
        );
    }
}

export default TwoColumenedBlock;
