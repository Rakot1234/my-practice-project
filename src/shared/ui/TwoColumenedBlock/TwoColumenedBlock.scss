@import '../../../styles/variables.scss';

$block: two-columened-block;

.#{$block} {
    display: flex;
    flex: 1 1 auto;
    justify-content: space-between;

    &__left {
        width: 44.8718%;
    }

    &__right {
        width: 51%;
    }
};
