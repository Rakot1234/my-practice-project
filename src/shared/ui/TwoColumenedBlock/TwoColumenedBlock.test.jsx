import React from 'react';
import { shallow } from 'enzyme';
import TwoColumenedBlock from './TwoColumenedBlock';

describe('<TwoColumenedBlock />', () => {
    it('Should render component', () => {
        const wrapper = shallow(<TwoColumenedBlock />);
        expect(wrapper).toMatchSnapshot();
    });
});
