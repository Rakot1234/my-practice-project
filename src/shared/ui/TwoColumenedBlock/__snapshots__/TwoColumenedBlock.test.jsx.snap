// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TwoColumenedBlock /> Should render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TwoColumenedBlock />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="two-columened-block__left"
        />,
        <div
          className="two-columened-block__right"
        />,
      ],
      "className": "two-columened-block",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "two-columened-block__left",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "two-columened-block__right",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="two-columened-block__left"
          />,
          <div
            className="two-columened-block__right"
          />,
        ],
        "className": "two-columened-block",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "two-columened-block__left",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "two-columened-block__right",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
